### ISSUE TITLE
[bug]Can not sync the chain

### ISSUE THREAD

**Issue by AndyChen1984** [filecoin-project/go-filecoin#2732]
**Describe the bug**
My node can not sync the chain ,and always has on block:
root@localhost:~/filecoin-proof-parameters# go-filecoin chain ls  
zDPWYqFCu9VdMN9pUSSttbjJsxd2sTJGZjXv8AGZvquxnsMRFh9e

But the log shows " INFO chain.sync: chain synced default_syncer.go:135"

**To Reproduce**
rm -rf ~/.filecoin
go-filecoin init --devnet-user --genesisfile=https://genesis.user.kittyhawk.wtf/genesis.car
go-filecoin daemon &

**Expected behavior**
go-filecoin chain ls  
show many blocks

**Screenshots**
23:32:41.064 WARNI /fil/hello: failed to send hello handshake to peer <peer.ID Qm*CM673T>: protocol not supported hello.go:167
23:33:02.736  INFO chain.sync: syncing the chain, currently at block height 68477 default_syncer.go:147
23:33:32.659  INFO    autonat: Discovered AutoNAT peer QmXq6XEYeEmUzBFuuKbVEGgxEpVD4xbSkG2Rhek6zkFMp4 notify.go:33

23:33:56.417  INFO chain.sync: syncing the chain, currently at block height 66160 default_syncer.go:147
23:34:46.622  INFO chain.sync: syncing the chain, currently at block height 63436 default_syncer.go:147
23:34:48.384  INFO    autonat: Discovered AutoNAT peer Qmd6xrWYHsxivfakYRy6MszTpuAiEoFbgE1LWw4EvwBpp4 notify.go:33
23:35:33.104  INFO chain.sync: syncing the chain, currently at block height 60374 default_syncer.go:147
23:35:35.330  INFO    autonat: Discovered AutoNAT peer QmXq6XEYeEmUzBFuuKbVEGgxEpVD4xbSkG2Rhek6zkFMp4 notify.go:33
23:35:41.236  INFO chain.sync: chain synced default_syncer.go:135
23:35:41.236  INFO       node: error handling blocks: { zDPWYqFD39ywezeED4uvDY9wzp4uEu4dYvL9pa2Z7QBnchm1bvNX } node.go:508
23:35:41.253  INFO chain.sync: syncing the chain, currently at block height 113053 default_syncer.go:147
23:35:46.841  INFO    autonat: Discovered AutoNAT peer QmXhxqTKzBKHA5FcMuiKZv8YaMPwpbKGXHRVZcFB2DX9XY notify.go:33
23:36:01.734  INFO chain.sync: syncing the chain, currently at block height 107814 default_syncer.go:147
23:36:08.716  INFO    autonat: Discovered AutoNAT peer QmXq6XEYeEmUzBFuuKbVEGgxEpVD4xbSkG2Rhek6zkFMp4 notify.go:33
23:36:50.494  INFO chain.sync: syncing the chain, currently at block height 105348 default_syncer.go:147
23:37:04.164  INFO chain.sync: chain synced default_syncer.go:135

root@localhost:~/filecoin-proof-parameters# go-filecoin show block zDPWYqFCu9VdMN9pUSSttbjJsxd2sTJGZjXv8AGZvquxnsMRFh9e
Block Details
Miner:  empty
Weight: 0.000
Height: 0
Nonce:  0

**Version information**

- Go: 
- Rust: 
- Cargo: 
- Commit: 

_go env_
```
<!-- go env -->
```
_cc_
```
<!-- $(go env | grep -E "^CC" |  sed -e 's/.*\=//' | sed -e 's/^"//' -e 's/"$//') --version -->
```

- Version: 

**Additional context**

**Comment by anorth** [comment]
Could you please fill out the version and commit information so that we can tell which code you're running?

**Comment by AndyChen1984** [comment]
> Could you please fill out the version and commit information so that we can tell which code you're running?

go-filecoin version
commit: 966fe75919755074c7c2724f2c9634c2b61d4212

thanks!

**Comment by anorth** [comment]
Thanks. That matches the 0.1.4 tag which is the current user devnet (not: that will change this week).

The "chain synced" message means the blocks have been downloaded, but not yet validated. Validation takes many hours after the download. I am not sure whether `show block` is expected to show you blocks that have not yet validated, but I could believe not. 

Have you waited for the chain to validate? One core of your CPU should be pretty busy until it finishes.

**Comment by evildido** [comment]
You need to wait the end of validation process.

Take a loot with the following command
watch -n 10 'go-filecoin show block $(go-filecoin chain head | head -n 1)'

**Comment by AndyChen1984** [comment]
> Thanks. That matches the 0.1.4 tag which is the current user devnet (not: that will change this week).
> 
> The "chain synced" message means the blocks have been downloaded, but not yet validated. Validation takes many hours after the download. I am not sure whether `show block` is expected to show you blocks that have not yet validated, but I could believe not.
> 
> Have you waited for the chain to validate? One core of your CPU should be pretty busy until it finishes.

1、yes the cpu is busy,top command shows 98%
"Validation takes many hours" means validating one block or all the blocks?
Using like “merkle tree” ?

2、Has  user devnet version changed? My node can not bootstrap today.
"
00:09:35.782 ERROR net.bootst: got error trying to connect to bootstrap node {ID:<peer.ID Qm*QvwuiV> Addrs:[/dns4/user.kittyhawk.wtf/tcp/9003]}: dial attempt failed: <peer.ID Qm*NJ5bTh> --> <peer.ID Qm*QvwuiV> dial attempt failed: dial tcp4 3.90.230.176:9003: connect: connection refused bootstrap.go:130"

And I find on"https://github.com/filecoin-project/go-filecoin","RADEME.md"shows the user dev net is on version "0.1.4",but the link directs  to  "test-devnet-0.2.0".

**Comment by AndyChen1984** [comment]
> You need to wait the end of validation process.
> 
> Take a loot with the following command
> watch -n 10 'go-filecoin show block $(go-filecoin chain head | head -n 1)'

Sorry,my node failed to boot today.I will report the log after I resolved the boot problem.
Thanks.

**Comment by anorth** [comment]
Validation involves replaying all the messages on the blockchain in order to build the state tree (as well as checking hashes etc). This replay consumes much of the time. We're investigating speed-ups to it now.

**Comment by anorth** [comment]
We have re-released the user devnet, so please connect to the `0.2.2` network

**Comment by AndyChen1984** [comment]
> We have re-released the user devnet, so please connect to the `0.2.2` network

I got it.Thanks

**Comment by rosalinekarr** [comment]
@AndyChen1984, did retrying with the `0.2.2` resolve your issue?

**Comment by AndyChen1984** [comment]
> @AndyChen1984, did retrying with the `0.2.2` resolve your issue?

I have two nodes:
One what I install from binary has synced the chain successfully.
Another one that I install from source can not bootstrap. "go-filecoin swarm peers" show nopeer.
"go-filecoin version" show null.

Thanks for reply

**Comment by rosalinekarr** [comment]
@AndyChen1984, are you still building from commit `966fe75919755074c7c2724f2c9634c2b61d4212` or another one?

**Comment by AndyChen1984** [comment]
> @AndyChen1984, are you still building from commit `966fe75919755074c7c2724f2c9634c2b61d4212` or another one?

no,I use the tag 0.2.2. But for rust-  fill-proofs,I update to master because of a compile bug.

The problem still exist. 
"go-filecoin swarm peers" show nopeer.
"go-filecoin version" show null.

screen log:

root@localhost:~/go_src/src/github.com/filecoin-project/go-filecoin/proofs/rust-fil-proofs# 12:30:39.151  INFO    autonat: Discovered AutoNAT peer QmZGDLdQLUTi7uYTNavKwCd7SBc5KMfxzWxAyvqRQvwuiV notify.go:33
12:30:53.920 WARNI /fil/hello: failed to send hello handshake to peer QmZGDLdQLUTi7uYTNavKwCd7SBc5KMfxzWxAyvqRQvwuiV: stream reset hello.go:179
12:31:02.128 WARNI /fil/hello: failed to send hello handshake to peer Qmd6xrWYHsxivfakYRy6MszTpuAiEoFbgE1LWw4EvwBpp4: stream reset hello.go:179
12:31:07.125  INFO    autonat: Discovered AutoNAT peer QmZGDLdQLUTi7uYTNavKwCd7SBc5KMfxzWxAyvqRQvwuiV notify.go:33


root@localhost:~/go_src/src/github.com/filecoin-project/go-filecoin# git log
commit fcf56acdc98a2ea4fc4d0b62878a9302f51af869 (HEAD, tag: test-devnet-0.2.2, tag: 0.2.2, origin/feat/stage-test-devnet-0.2.1)
Author: frrist <forrest@protocol.ai>

root@localhost:~/go_src/src/github.com/filecoin-project/go-filecoin/proofs/rust-fil-proofs# git log
commit ac53936458961a2b25f90aa53fdf607d65f9b008 (HEAD -> master, tag: ac53936458961a2b, origin/master, origin/HEAD)
Merge: 2673ab7 a311b9e
Author: porcuquine <1746729+porcuquine@users.noreply.github.com>
Date:   Tue May 14 00:30:02 2019 -0700

**Comment by AndyChen1984** [comment]
@rosalinekarr the install command will insert the git commit info fail. It caused my node bootstrap fail.
I use cp command after build,the problem is fixed.

**Comment by anorth** [comment]
I'm sorry this issue has languished. We recently released a new network 0.3.2. Please upgrade to the latest code and try again. Feel free to re-open this issue if you continue to have trouble.

**Comment by voidao** [comment]
@AndyChen1984 Seems like I met the same issue.

**_I found it might be related to the `correct code` restriction_**, as instructed in the [Getting-Started](https://github.com/filecoin-project/go-filecoin/wiki/Getting-Started)

> **Installing from source**
Use these steps to install filecoin:
1. Build dependencies
...

2. Find the latest git tag for the user devnet from the project README by clicking the badge (user devnet etc) that takes you to the release page (e.g., "0.3.2").

3. Ensure you have the correct code for joining the user devnet:

cd $GOPATH/src/github.com/filecoin-project/go-filecoin
git fetch origin
git checkout $USER_DEVNET_TAG

---
**To sum up, making sure that you have the correct code for the user devnet, which is 0.3.2 for now.**

$ git status
HEAD detached at 0.3.2
nothing to commit, working tree clean
$ git log
commit 0edc3e9c95f198cc5a20c3f94badb6d86160d7cc (HEAD, tag: test-devnet-0.3.2, tag: test-devnet-0.3.1, tag: 0.3.2, origin/release-0.3.1)


> Good luck ~
